File: Strings/count_And_Say.java
class Solution {
    public String countAndSay(int n) {
        String str="11";
        if(n==1){
            return "1";
        }
        for (int j=0;j<n-2;j++){
            int count=0;
            int i=0;
            char s=str.charAt(0);
            String s1="";
            while(i<str.length()){
                    if(str.charAt(i)==s){
                        count++;
                    }else{
                        s1=s1+count+s;
                        s=str.charAt(i);
                        count=0;
                        continue;
                    }
                    
                    if(i==str.length()-1){
                        s1=s1+count+s;
                        s=str.charAt(i);
                    }
                    i++;
                }
                str=s1;
        }
        return str;
    }
}

File: strings/count and say.cpp
class Solution {
    private:
    string say(string s){
        string result="";
        for(int i=0;i<s.length();i++){
            int count=1;
            while(i<s.length()-1&& s[i]==s[i+1]){
                count++;
                i++;
            }
            string str=to_string(count);
            result+=str;
            result+=s[i];
        }
        return result;
    }
public:
    string countAndSay(int n) {
        string s="1";
        if(n==1)
        return s;
        for(int i=2;i<=n;i++){
            string str=say(s);
            s=str;
        }
        return s;
    }
};